File: library/src/androidTest/java/com/raizlabs/android/dbflow/test/structure/ForeignModel2.java
package com.raizlabs.android.dbflow.test.structure;

import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.ForeignKey;
import com.raizlabs.android.dbflow.annotation.ForeignKeyReference;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.test.TestDatabase;
import com.raizlabs.android.dbflow.test.structure.autoincrement.TestModelAI;

@Table(databaseName = TestDatabase.NAME)
public class ForeignModel2 extends TestModel1 {

    @Column
    @ForeignKey(references = {@ForeignKeyReference(columnName = "testmodel_id",
            columnType = Long.class, foreignColumnName = "id")})
    TestModelAI testModelAI;
}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/structure/foreignkey/ForeignModel2.kt
package com.raizlabs.android.dbflow.structure.foreignkey

import com.raizlabs.android.dbflow.annotation.Column
import com.raizlabs.android.dbflow.annotation.ForeignKey
import com.raizlabs.android.dbflow.annotation.ForeignKeyReference
import com.raizlabs.android.dbflow.annotation.Table
import com.raizlabs.android.dbflow.TestDatabase
import com.raizlabs.android.dbflow.structure.TestModel1
import com.raizlabs.android.dbflow.structure.autoincrement.TestModelAI

@Table(database = TestDatabase::class)
class ForeignModel2 : TestModel1() {

    @Column
    @ForeignKey(references = arrayOf(
        ForeignKeyReference(columnName = "testmodel_id", foreignKeyColumnName = "id")))
    var testModelAI: TestModelAI? = null
}
